Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
-// Virtual method:  Create
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Makes the FrameMan object ready for use, which is to be used with
-//                  SettingsMan first.
-
-int FrameMan::Create()
-{
-    m_pFrameTimer = new Timer();
-
-    // Init allegro's graphics
-    set_color_depth(m_BPP);
-
-//    GFX_MODE_LIST *pList = get_gfx_mode_list(GFX_DIRECTX_ACCEL);
-//    destroy_gfx_mode_list(pList);
-
-    // Refuse windowed multiplier if the resolution is too high
-    if (m_ResX > 1024)
-        m_NxWindowed = 1;
-/*
-    // Force double virtual fullscreen res if the res is too high
-    if (m_ResX >= 1280)// && m_Fullscreen && m_NxFullscreen == 1)
-    {
-        m_NxFullscreen = 2;
-        m_NewResX = m_ResX /= m_NxFullscreen;
-        m_NewResY = m_ResY /= m_NxFullscreen;
-    }
-    // Not oversized resolution
-    else
-        m_NxFullscreen = 1;
-*/
-
-	int fullscreenGfxDriver = GFX_AUTODETECT_FULLSCREEN;
-	int windowedGfxDriver = GFX_AUTODETECT_WINDOWED;
-
-    fullscreenGfxDriver = GFX_DIRECTX_ACCEL;
-
-	if (g_SettingsMan.ForceSoftwareGfxDriver()) 
-		fullscreenGfxDriver = GFX_DIRECTX_SOFT;
-	if (g_SettingsMan.ForceSafeGfxDriver()) 
-		fullscreenGfxDriver = GFX_DIRECTX_SAFE;
-	if (g_SettingsMan.ForceOverlayedWindowGfxDriver()) 
-		windowedGfxDriver = GFX_DIRECTX_OVL;
-	if (g_SettingsMan.ForceNonOverlayedWindowGfxDriver()) 
-		windowedGfxDriver = GFX_DIRECTX_WIN;
-	if (g_SettingsMan.ForceVirtualFullScreenGfxDriver()) 
-		windowedGfxDriver = GFX_DIRECTX_WIN_BORDERLESS;
-
-
-    if (set_gfx_mode(m_Fullscreen ? fullscreenGfxDriver : windowedGfxDriver, m_Fullscreen ? m_ResX * m_NxFullscreen : m_ResX * m_NxWindowed, m_Fullscreen ? m_ResY * m_NxFullscreen : m_ResY * m_NxWindowed, 0, 0) != 0)
-    {
-		g_ConsoleMan.PrintString("Failed to set gfx mode, trying different windowed scaling.");
-
-		// If player somehow managed to set up a windowed 2X mode and then set a resoultion higher than physical, then disable 2X resolution
-		m_NxWindowed = m_NxWindowed == 2 ? 1 : m_NxWindowed;
-		if (set_gfx_mode(m_Fullscreen ? fullscreenGfxDriver : windowedGfxDriver, m_Fullscreen ? m_ResX * m_NxFullscreen : m_ResX * m_NxWindowed, m_Fullscreen ? m_ResY * m_NxFullscreen : m_ResY * m_NxWindowed, 0, 0) != 0)
-	    {
-			g_ConsoleMan.PrintString("Failed to set gfx mode, trying different fullscreen scaling.");
-
-			// TODO: this is whack if we're attempting windowed
-			// Oops, failed to set fullscreen mode, try a different fullscreen scaling
-			m_NewNxFullscreen = m_NxFullscreen = m_NxFullscreen == 1 ? 2 : 1;
-			if (set_gfx_mode(m_Fullscreen ? fullscreenGfxDriver : windowedGfxDriver, m_Fullscreen ? m_ResX * m_NxFullscreen : m_ResX * m_NxWindowed, m_Fullscreen ? m_ResY * m_NxFullscreen : m_ResY * m_NxWindowed, 0, 0) != 0)
+
+		//m_StoreNetworkBackBuffer = false;
+		if (m_StoreNetworkBackBuffer) {
```

Comment:
Clean out the commented code and move this multiplayer business into a separate private method. m_StoreNetworkBackBuffer is the same as `IsInMultiplayerMode()`, which is basically saying, "this is a multiplayer server". Honestly it'd be nice to be consistent and call `IsInMultiplayerMode()` everywhere to avoid confusion. Also rename IsInMultiplayerMode() to `IsMultiplayerServer` or something across the board, but that's a totally separate piece of work for a totally separate issue.

Also, If the above acquire_bitmap stuff works then I'd like the multiplayer stuff there put into this method (so IsInMultiplayerMode check would be made inside the method and either option would be done appropriately) so it's all grouped together in one multiplayer handling method.
The splitscreen stuff being couched in, essentially, a `!IsInMultiplayerMode()` check is because the server doesn't draw screen dividing lines for multiple players.